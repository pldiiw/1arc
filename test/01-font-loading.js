'use strict';

const log = console.log;
const hex = n => n.toString(16).toUpperCase();
const binToDec = s => parseInt(s, 2);

const engine = require('engine').initialize();
const fontLoadedEngine = require('engine').loadFont(engine);

const memory = fontLoadedEngine.memory;
const loadedSpriteSet = memory.slice(0, 79 * 8 + 1);

[...Array(16).keys()]
  .map(v => hex(v))
  .map(v => {
    log(`${v}: ${loadedSpriteSet.slice(v * 5, (v + 1) * 5).join(' ')}`);
  });

