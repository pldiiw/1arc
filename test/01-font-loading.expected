0: F0 90 90 90 F0
1: 20 60 20 20 70
2: F0 10 F0 80 F0
3: F0 10 F0 10 F0
4: 90 90 F0 10 10
5: F0 80 F0 10 F0
6: F0 80 F0 90 F0
7: F0 10 20 40 40
8: F0 90 F0 90 F0
9: F0 90 F0 10 F0
A: F0 90 F0 90 90
B: E0 90 E0 90 E0
C: F0 80 80 80 F0
D: E0 90 90 90 E0
E: F0 80 F0 80 F0
F: F0 80 F0 80 80
